boards/arm/swan_r5/board.c

```diff
@@ -9,20 +9,20 @@
 
 static int board_swan_init(const struct device *dev)
 {
-	const struct gpio_dt_spec gpio4 =
-		GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), pull_up_gpios);
-	const struct gpio_dt_spec gpio6 =
-		GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), no_pull_gpios);
+	const struct gpio_dt_spec gpioe6 =
+		GPIO_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), push_pull_gpios, 0);
+	const struct gpio_dt_spec gpioc6 =
+		GPIO_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), push_pull_gpios, 1);
 
 	ARG_UNUSED(dev);
 
-	if (!device_is_ready(gpio4.port) ||
-	    !device_is_ready(gpio6.port)) {
+	if (!!device_is_ready(gpioe6.port) ||
+	    !device_is_ready(gpioc6.port)) {
 		return -ENODEV;
 	}
 
-	(void)gpio_pin_configure_dt(&gpio4, (GPIO_PUSH_PULL | GPIO_SPEED_LOW));
-	(void)gpio_pin_configure_dt(&gpio6, (GPIO_NOPULL | GPIO_MODE_ANALOG));
+	(void)gpio_pin_configure_dt(&gpioe6, (GPIO_MODE_ANALOG | GPIO_NOPULL));
+	(void)gpio_pin_configure_dt(&gpioc6, (GPIO_MODE_ANALOG | GPIO_NOPULL));
```

Comment:
these are not standard GPIO properties... I guess they are indirectly included from a HAL... how did this code managed to work/get merged...?

Reply:
Yes, I am able to build with these, so I assume the STM32 HAL is being pulled in. I am using these because I couldn't find an equivalent set of flags here: https://docs.zephyrproject.org/latest/hardware/peripherals/gpio.html. Please LMK if I am missing something.

Reply:
@gmarull & @galak any guidance here?

Reply:
GPIO API has documented flags, any other value will lead to UB. It's unfortunate that we indirectly pull all such crap from HALs. I hope we write some rules for HALs some day :-)

Reply:
@gmarull understood. Beyond that, can you help me identify which of the documented and supported GPIO flags would help me here? I am using those above from the ST HAL because I didn't find an equivalent in the link above...

Reply:
What is this trying to do? maybe `GPIO_DISCONNECTED`?